ocular/src/chain/client/query.rs

```diff
@@ -1,113 +1,86 @@
 use tonic::transport::Channel;
 
-use super::ChainClient;
-
-pub type AuthQueryClient = auth::query_client::QueryClient<Channel>;
-pub type BankQueryClient = bank::query_client::QueryClient<Channel>;
-pub type DistributionQueryClient = distribution::query_client::QueryClient<Channel>;
-pub type EvidenceQueryClient = evidence::query_client::QueryClient<Channel>;
-pub type GovQueryClient = gov::query_client::QueryClient<Channel>;
-pub type MintQueryClient = mint::query_client::QueryClient<Channel>;
-pub type ParamsQueryClient = params::query_client::QueryClient<Channel>;
-pub type SlashingQueryClient = slashing::query_client::QueryClient<Channel>;
-pub type StakingQueryClient = staking::query_client::QueryClient<Channel>;
-
-impl ChainClient {
-    // Auth queries
-    // TODO: Refractor code accross grpc clients by having keystore implement Sync (https://github.com/PeggyJV/ocular/pull/53#discussion_r880698565)
-    pub async fn get_auth_query_client(&mut self) -> Result<AuthQueryClient, ChainClientError> {
-        let mut result: Result<AuthQueryClient, ChainClientError> =
-            Err(TxError::Broadcast(String::from("Client connection never attempted.")).into());
-
-        // Get grpc address randomly each time; shuffles on failures
-        for _i in 0u8..self.connection_retry_attempts + 1 {
-            // Attempt to use last healthy (or manually set) endpoint if it exists (in config)
-            let endpoint: String = if !self.config.grpc_address.is_empty() {
-                self.config.grpc_address.clone()
-            } else {
-                // Get a random endpoint from the cache
-                match self.get_random_grpc_endpoint().await {
-                    Ok(endpt) => endpt,
-                    Err(err) => return Err(GrpcError::MissingEndpoint(err.to_string()).into()),
-                }
-            };
-
-            result = AuthQueryClient::connect(endpoint.clone())
-                .await
-                .map_err(|e| GrpcError::Connection(e).into());
-
-            // Return if result is valid client, or increment failure in cache if being used
-            if result.is_ok() {
-                // Reset consecutive failed connections to 0
-                self.cache
-                    .as_mut()
-                    .unwrap()
-                    .grpc_endpoint_cache
-                    .add_item(endpoint.clone(), 0)?;
-
-                // Update config to last healthy grpc connection address
-                self.config.grpc_address = endpoint.clone();
-
-                break;
-            } else if result.is_err() && self.cache.is_some() {
-                // Don't bother updating config grpc address if it fails, it'll be overriden upon a successful connection
-                let _res = self
-                    .cache
-                    .as_mut()
-                    .unwrap()
-                    .grpc_endpoint_cache
-                    .increment_failed_connections(endpoint)?;
-            }
-        }
```

Comment:
Unless I'm mistaken don't these changes wipe out the grpc endpoint caching for all clients we implemented earlier?